File: api/src/main/java/com/sourceforgery/swedbank/CPN.java
package com.sourceforgery.swedbank;

import java.util.Map;

public class CPN {
    private final String avv;
    private final String from;
    private final String pan;
    private final String expiryMonth;
    private final String expiryYear;

    private CPN(final Map<String, String> data) {
        avv = data.get("AVV");
        from = data.get("From");
        pan = data.get("PAN");
        expiryMonth = data.get("ExpiryMonth");
        expiryYear = data.get("ExpiryYear");
    }

    public static CPN from(final Map<String, String> data) {
        return new CPN(data);
    }
}

File: api/src/main/java/com/sourceforgery/swedbank/CPN.kt
package com.sourceforgery.swedbank

class CPN private constructor(data: Map<String, String>) {
    private val avv: String
    private val from: String
    private val pan: String
    private val expiryMonth: String
    private val expiryYear: String

    init {
        avv = data["AVV"]
        from = data["From"]
        pan = data["PAN"]
        expiryMonth = data["ExpiryMonth"]
        expiryYear = data["ExpiryYear"]
    }

    companion object {

        fun from(data: Map<String, String>): CPN {
            return CPN(data)
        }
    }
}
